addons/test_mail/tests/test_mail_message.py

```diff
@@ -508,3 +509,77 @@ def test_mail_message_access_create_wo_parent_access(self):
 
         self.assertTrue(new_mail)
         self.assertEqual(new_msg.parent_id, message)
+
+
+@tagged("mail_message")
+class TestMessageLinks(MailCommon, HttpCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.user_employee_1 = mail_new_test_user(cls.env, login='tao1', groups='base.group_user', name='Tao Lee')
+        cls.public_channel = cls.env['discuss.channel'].channel_create(name='Public Channel1', group_id=None)
+        cls.private_group = cls.env['discuss.channel'].create_group(partners_to=cls.user_employee_1.partner_id.ids, name="Group")
```

Comment:
This actually belongs to discuss tests

Reply:
I get your point that it is using `discuss` specific models. But it isn't using Discuss only. This tests other models aswell. So, IMO all the tests testing a common feature should kept in a single place if possible.